### DISCUSSION TITLE
为什么纯IP的DOH也能正常工作？

### DISCUSSION THREAD

**Discussion by 57382** [IrineSistiana/mosdns#681] [Q&A] [answered]
https://github.com/AdguardTeam/AdGuardHome/issues/5217
根据这里的描述，DOH缺少域名地址会报证书错误，

为什么我用
      upstreams:
        - addr: https://1.1.1.1/dns-query
可以正常工作？

我没开启MOS的忽略证书错误选项。

**Comment by sensec** [comment]
google和cloudflare有IP别名的证书，可以直接写IP形式。

**Comment by sensec** [comment, marked as answer]
你可以访问 https://one.one.one.one/ 看一下cloudflare-dns的证书，包含以下备用名：
DNS 名称: cloudflare-dns.com
DNS 名称: *.cloudflare-dns.com
DNS 名称: one.one.one.one
IP 地址: 1.0.0.1
IP 地址: 1.1.1.1
IP 地址: 162.159.36.1
IP 地址: 162.159.46.1
IP 地址: 2606:4700:4700::1001
IP 地址: 2606:4700:4700::1111
IP 地址: 2606:4700:4700::64
IP 地址: 2606:4700:4700::6400

**Reply by 57382** [reply]
> 你可以访问 https://one.one.one.one/ 看一下cloudflare-dns的证书，包含以下备用名： DNS 名称: cloudflare-dns.com DNS 名称: *.cloudflare-dns.com DNS 名称: one.one.one.one IP 地址: 1.0.0.1 IP 地址: 1.1.1.1 IP 地址: 162.159.36.1 IP 地址: 162.159.46.1 IP 地址: 2606:4700:4700::1001 IP 地址: 2606:4700:4700::1111 IP 地址: 2606:4700:4700::64 IP 地址: 2606:4700:4700::6400



谁都不选就选你。这位老师答的好。
